app/controllers/application_controller.rb

```diff
@@ -127,4 +127,12 @@ def school_root_url_for(user)
       schools_path
     end
   end
+
+  def show_parent_carer_pupil_banner?
+    @show_parent_carer_pupil_banner = if current_user.id.present?
+                                        false
+                                      else
+                                        true
+                                      end
```

Comment:
```suggestion
  def show_parent_carer_pupil_banner?
    @show_parent_carer_pupil_banner = current_user&.new_record?
```

You can omit the conditional here and check directly whether the user is persisted.